app/src/main/java/com/pajato/android/gamechat/exp/CheckersFragment.java

```diff
@@ -69,107 +125,355 @@
                 PorterDuff.Mode.SRC_ATOP);
     }
 
+    /** Handle taking the foreground by updating the UI based on the current experience. */
+    @Override public void onResume() {
+        // Determine if there is an experience ready to be enjoyed.  If not, hide the layout and
+        // present a spinner.  When an experience is posted by the app event manager, the game can
+        // be shown
+        super.onResume();
+        resume();
+    }
+
+    /** Return a default, partially populated, Checkers experience. */
+    protected void createExperience(final Context context, final List<Account> playerAccounts) {
+        // Setup the default key, players, and name.
+        String key = getExperienceKey();
+        List<Player> players = getDefaultPlayers(context, playerAccounts);
+        String name1 = players.get(0).name;
+        String name2 = players.get(1).name;
+        long tstamp = new Date().getTime();
+        String name = String.format(Locale.US, "%s vs %s on %s", name1, name2, tstamp);
+
+        // Set up the default group and room keys, the owner id and return the value.
+        String groupKey = GroupManager.instance.getGroupKey();
+        String roomKey = RoomManager.instance.getRoomKey(groupKey);
+        String id = getOwnerId();
+        // TODO: DEFINE LEVEL INT ENUM VALUES - this is passing "0" for now
+        Checkers model = new Checkers(key, id, 0, name, tstamp, groupKey, roomKey, players);
+        ExperienceManager.instance.createExperience(model);
+    }
+
+    /** Return a possibly null list of player information for a checkers experience (always 2 players) */
+    protected List<Account> getPlayers(final Dispatcher<ExpFragmentType, ExpProfile> dispatcher) {
+        // Determine if this is an offline experience in which no accounts are provided.
+        Account player1 = AccountManager.instance.getCurrentAccount();
+        if (player1 == null) return null;
+
+        // This is an online experience.  Use the current signed in User as the first player.
+        List<Account> players = new ArrayList<>();
+        players.add(player1);
+
+        // Determine the second account, if any, based on the room.
+        String key = dispatcher.roomKey;
+        Room room = key != null ? RoomManager.instance.roomMap.get(key) : null;
+        int type = room != null ? room.type : -1;
+        switch (type) {
```

Comment:
As I look at this detritus (which is mine) two thoughts arise:  first, this segment should be removed and added back in if and when it becomes clear what needs to be done; second, much as we model the two players on Firebase as a List<Player> it strikes me that we could model a board for chess and checkers as two lists of player pieces, (n lists for n players), something like: `List<CheckerPlayerPiece> redPieces` and `List<ChessPlayerPiece> blackPieces` where *PlayerPiece is a POJO that encapsulates color, position and piece type.